components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2RegistrationEventPayloadBuilder.java

```diff
@@ -0,0 +1,236 @@
+
+                    emailAddress =
+                            userStoreManager.getUserClaimValue(userName, FrameworkConstants.EMAIL_ADDRESS_CLAIM,
+                                    UserCoreConstants.DEFAULT_PROFILE);
+
+                }
+                if (StringUtils.isEmpty(givenName)) {
+
+                    givenName = userStoreManager.getUserClaimValue(userName, FIRST_NAME_CLAIM_URI,
+                            UserCoreConstants.DEFAULT_PROFILE);
+                }
+
+                if (StringUtils.isEmpty(lastName)) {
+
+                    lastName = userStoreManager.getUserClaimValue(userName, LAST_NAME_CLAIM_URI,
+                            UserCoreConstants.DEFAULT_PROFILE);
+
+                }
+            } catch (UserStoreException e) {
+                throw new IdentityEventException(
+                        "Error while extracting user claims for the user : " + user.getId(), e);
+            }
+
+            UserClaim emailAddressUserClaim = new UserClaim(FrameworkConstants.EMAIL_ADDRESS_CLAIM, emailAddress);
+            UserClaim givenNameUserClaim = new UserClaim(FIRST_NAME_CLAIM_URI, givenName);
+            UserClaim lastNameUserClaim = new UserClaim(LAST_NAME_CLAIM_URI, lastName);
+
+            userClaims.add(emailAddressUserClaim);
+            userClaims.add(givenNameUserClaim);
+            userClaims.add(lastNameUserClaim);
+            user.setClaims(userClaims);
+        } else {
+            enrichUser(userStoreManager, userName, user);
+        }
+    }
+
+    private List<String> getRegistrationMethods(String[] internalRoles) {
+
+        if (internalRoles == null || internalRoles.length == 0) {
+            return null;
+        }
+        List<String> roles = Arrays.asList(internalRoles);
+
+        List<String> registrationMethods = new ArrayList<>();
+        Claim emailVerifyTemporaryClaim = Utils.getEmailVerifyTemporaryClaim();
+
+        if (roles.contains(IdentityRecoveryConstants.SELF_SIGNUP_ROLE)) {
+            registrationMethods.add(UserOnboardedMethod.SELF_SIGNUP.name());
+        } else if (emailVerifyTemporaryClaim != null &&
+                IdentityRecoveryConstants.ASK_PASSWORD_CLAIM.equals(emailVerifyTemporaryClaim.getClaimUri())) {
+            registrationMethods.add(UserOnboardedMethod.USER_INVITE.name());
+        } else {
+            registrationMethods.add(UserOnboardedMethod.ADMIN_INITIATED.name());
+        }
+
+        return registrationMethods;
+    }
+
+    private void addRoles(Map<String, Object> properties, User user) {
+
+        if (!properties.containsKey(IdentityEventConstants.EventProperty.ROLE_LIST)) {
+            return;
+        }
+        String[] roleList = (String[]) properties.get(IdentityEventConstants.EventProperty.ROLE_LIST);
+
+        for (String role : roleList) {
+            user.addRole(role);
+        }
+    }
+
+    private static void enrichUser(UserStoreManager userStoreManager, String domainQualifiedUserName, User user)
```

Comment:
We should only include the claims the user register with. Don't pull claims from the user store